e2e/multinet_test.go

```diff
@@ -0,0 +1,268 @@
+	sanboxImage = "weaveworks/ignite:dev"
+	kernelImage = "weaveworks/ignite-kernel:5.4.108"
+	vmImage     = "weaveworks/ignite-ubuntu"
+)
+
+func startAsyncVM(t *testing.T, intfs []string) (*operations.VMChannels, string) {
+
+	assert.Assert(t, e2eHome != "", "IGNITE_E2E_HOME should be set")
+	igniteUtil.GenericCheckErr(providers.Populate(ignite.Preload))
+
+	_ = igniteDocker.SetDockerRuntime()
+	_ = igniteDocker.SetDockerNetwork()
+
+	providers.RuntimeName = runtime.RuntimeDocker
+	providers.NetworkPluginName = network.PluginDockerBridge
+	_ = providers.Populate(ignite.Providers)
+
+	vm := providers.Client.VMs().New()
+	vm.Status.Runtime.Name = runtime.RuntimeDocker
+	vm.Status.Network.Plugin = network.PluginDockerBridge
+
+	ociRef, err := meta.NewOCIImageRef(sanboxImage)
+	if err != nil {
+		t.Fatalf("Failed to parse OCI image ref %s: %s", sanboxImage, err)
+	}
+	vm.Spec.Sandbox.OCI = ociRef
+
+	ociRef, err = meta.NewOCIImageRef(kernelImage)
+	if err != nil {
+		t.Fatalf("Failed to parse OCI image ref %s: %s", kernelImage, err)
+	}
+	vm.Spec.Kernel.OCI = ociRef
+	k, _ := operations.FindOrImportKernel(providers.Client, ociRef)
+	vm.SetKernel(k)
+
+	ociRef, err = meta.NewOCIImageRef(vmImage)
+	if err != nil {
+		t.Fatalf("Failed to parse OCI image ref %s: %s", vmImage, err)
+	}
+	img, err := operations.FindOrImportImage(providers.Client, ociRef)
+	if err != nil {
+		t.Fatalf("Failed to find OCI image ref %s: %s", ociRef, err)
+	}
+	vm.SetImage(img)
+
+	vm.Name = multinetVM
+	vm.Spec.SSH = &api.SSH{Generate: true}
+
+	_ = metadata.SetNameAndUID(vm, providers.Client)
+
+	for _, intf := range intfs {
+		vm.SetAnnotation(igniteConstants.IGNITE_INTERFACE_ANNOTATION+intf, "tc-redirect")
+	}
+
+	_ = providers.Client.VMs().Set(vm)
+
+	err = dmlegacy.AllocateAndPopulateOverlay(vm)
+	if err != nil {
+		t.Fatalf("Error AllocateAndPopulateOverlay: %s", err)
+	}
+
+	vmChans, err := operations.StartVMNonBlocking(vm, false)
+	if err != nil {
+		t.Fatalf("failed to start a VM: \n%q\n", err)
+	}
+
+	return vmChans, vm.GetUID().String()
+}
+
+// TestMultipleInterface tests that a VM's can be configured with more than 1 interface
+func TestOneExtraInterface(t *testing.T) {
+	assert.Assert(t, e2eHome != "", "IGNITE_E2E_HOME should be set")
+
+	igniteCmd := util.NewCommand(t, igniteBin)
+	dockerCmd := util.NewCommand(t, runtime.RuntimeDocker.String())
+
+	vmChans, vmID := startAsyncVM(t, []string{"foo"})
+
+	// Clean-up the following VM.
+	defer igniteCmd.New().
+		With("rm", "-f", multinetVM).
+		Run()
+
+	fooAddr := "aa:ca:e9:12:34:56"
+	dockerCmd.New().
+		With("exec", fmt.Sprintf("ignite-%s", vmID)).
+		With("ip", "link", "add", "foo", "type", "veth").
+		Run()
+
+	dockerCmd.New().
+		With("exec", fmt.Sprintf("ignite-%s", vmID)).
+		With("ip", "link", "set", "foo", "address", fooAddr).
+		Run()
+
+	// check that the VM has started before trying exec
+	if err := <-vmChans.SpawnFinished; err != nil {
+		t.Fatalf("failed to start a VM: \n%q\n", err)
+	}
```

Comment:
I like that these tests exercise the wait code that polls for the Firecracker Prometheus socket.